### PR TITLE
Use Quantity in ResourceInfo

### PR DESCRIPTION
**Pull request by marquiz**
Use Quantity instead of IntOrString in order to guarantee full
compatibility with the Kubernetes API. With this we also get CRD
validation.

Auto-generated code (deepcopy functions and crd.yaml) updated with
controller-gen v0.6.0:

  controller-gen object crd output:crd:dir=. paths=./pkg/apis/...
  mv topology.node.k8s.io_noderesourcetopologies.yaml manifests/crd.yaml

### CODE REVIEW

File: manifests/crd.yaml [hunk 1]
```diff
@@ -1,68 +1,113 @@
+
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    "api-approved.kubernetes.io": "https://github.com/kubernetes/enhancements/pull/1870"
```
Comment (AlexeyPerevalov) [inline manifests/crd.yaml:6]: as I remember at the time of initial commit, about a year ago this field: api-approved.kubernetes.io was mandatory. For controlller-gen it's not, right? Where we can see a reference?

File: manifests/crd.yaml [hunk 2]
```diff
@@ -1,68 +1,113 @@
...
     singular: noderesourcetopology
-    kind: NodeResourceTopology
-    shortNames:
-      - node-res-topo
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: NodeResourceTopology is a specification for a Foo resource
```
Comment (AlexeyPerevalov) [inline manifests/crd.yaml]: I'm not sure about Foo ;) maybe better to replace it to:
is a specification for a hardware resource

**Review by AlexeyPerevalov — COMMENTED** [review]
@marquiz, did you started a samples or real code (e.g. scheduler-plugin) which check this modification? If not  - I can.

**Comment by AlexeyPerevalov** [comment]
I checked it with scheduler-plugin, and made a PR https://github.com/kubernetes-sigs/scheduler-plugins/pull/214.

File: manifests/crd.yaml (hunk 1 above)
Comment (marquiz) [reply manifests/crd.yaml:6]: Good point, addressed in the newest version of the pr. I added a kubectl command to add the annotation

File: manifests/crd.yaml (hunk 2 above)
Comment (marquiz) [reply manifests/crd.yaml]: Hmm, these come directly from the code comments. Added/updated comments for all the exported api types

**Comment by marquiz** [comment]
Rebased and updated

**Comment by AlexeyPerevalov** [comment]
/lgtm
@swatisehgal what do you think?

**Pull request merged**